src/backend/base/langflow/api/v1/mcp.py

```diff
@@ -0,0 +1,295 @@
+from langflow.services.auth.utils import get_current_active_user
+from langflow.services.database.models import Flow, User
+from langflow.services.deps import get_db_service, get_session
+
+logger = logging.getLogger(__name__)
+if False:
+    logger.setLevel(logging.DEBUG)
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        handler.setLevel(logging.DEBUG)
+        formatter = logging.Formatter("[%(asctime)s][%(levelname)s] %(message)s")
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+
+    # Enable debug logging for MCP package
+    mcp_logger = logging.getLogger("mcp")
+    mcp_logger.setLevel(logging.DEBUG)
+    if not mcp_logger.handlers:
+        mcp_logger.addHandler(handler)
+
+    logger.debug("MCP module loaded - debug logging enabled")
+
+router = APIRouter(prefix="/mcp", tags=["mcp"])
+
+server = Server("langflow-mcp-server")
+
+# Create a context variable to store the current user
+current_user_ctx: ContextVar[User] = ContextVar("current_user_ctx")
+
+
+def json_schema_from_flow(flow: Flow) -> dict:
+    """Generate JSON schema from flow input nodes."""
+    # Get the flow's data which contains the nodes and their configurations
+    flow_data = flow.data if flow.data else {}
+
+    graph = Graph.from_payload(flow_data)
+    input_nodes = [vertex for vertex in graph.vertices if vertex.is_input]
+
+    properties = {}
+    required = []
+    for node in input_nodes:
+        node_data = node.data["node"]
+        template = node_data["template"]
+
+        for field_name, field_data in template.items():
+            if field_data != "Component" and field_data.get("show", False) and not field_data.get("advanced", False):
+                field_type = field_data.get("type", "string")
+                properties[field_name] = {
+                    "type": field_type,
+                    "description": field_data.get("info", f"Input for {field_name}"),
+                }
+                # Update field_type in properties after determining the JSON Schema type
+                if field_type == "str":
+                    field_type = "string"
+                elif field_type == "int":
+                    field_type = "integer"
+                elif field_type == "float":
+                    field_type = "number"
+                elif field_type == "bool":
+                    field_type = "boolean"
+                else:
+                    logger.warning(f"Unknown field type: {field_type} defaulting to string")
+                    field_type = "string"
+                properties[field_name]["type"] = field_type
+
+                if field_data.get("required", False):
+                    required.append(field_name)
+
+    return {"type": "object", "properties": properties, "required": required}
+
+
+@server.list_prompts()
+async def handle_list_prompts():
+    return []
+
+
+@server.list_resources()
+async def handle_list_resources():
+    return []
```

Comment:
Should we expose the files endpoints here? Maybe some others..

Reply:
added files